dashboard/src/components/Header/ContextSelector.tsx

```diff
@@ -0,0 +1,134 @@
+import { push } from "connected-react-router";
+import React, { useEffect, useRef, useState } from "react";
+import { useDispatch } from "react-redux";
+
+import { angleIcon, ClarityIcons, clusterIcon, fileGroupIcon } from "@clr/core/icon-shapes";
+import { IClustersState } from "../../reducers/cluster";
+import { CdsButton, CdsIcon } from "../Clarity/clarity";
+import useOutsideClick from "../js/hooks/useOutsideClick/useOutsideClick";
+
+import Column from "components/js/Column";
+import { definedNamespaces } from "shared/Namespace";
+import { app } from "shared/url";
+import Row from "../js/Row";
+import "./ContextSelector.css";
+
+ClarityIcons.addIcons(clusterIcon, fileGroupIcon, angleIcon);
+
+export interface IContextSelectorProps {
+  clusters: IClustersState;
+  defaultNamespace: string;
+  fetchNamespaces: () => void;
+  createNamespace: (ns: string) => Promise<boolean>;
+  getNamespace: (ns: string) => void;
+  setNamespace: (ns: string) => void;
+}
+
+function ContextSelector({
+  clusters,
+  defaultNamespace,
+  fetchNamespaces,
+  createNamespace,
+  getNamespace,
+  setNamespace,
+}: IContextSelectorProps) {
+  const dispatch = useDispatch();
+  const [open, setOpen] = useState(false);
+  const currentCluster = clusters.clusters[clusters.currentCluster];
+  const namespaceSelected = currentCluster.currentNamespace || defaultNamespace;
+  const [cluster, setCluster] = useState(clusters.currentCluster);
+  const [namespace, setNS] = useState(namespaceSelected);
+  // Control when users click outside
+  const ref = useRef(null);
+  useOutsideClick(setOpen, [ref], open);
+
+  useEffect(() => {
+    fetchNamespaces();
+    if (namespaceSelected !== definedNamespaces.all) {
+      getNamespace(namespaceSelected);
+    }
+  }, [fetchNamespaces, namespaceSelected, getNamespace]);
+
+  const toggleOpen = () => setOpen(!open);
+  const selectCluster = (event: React.ChangeEvent<HTMLSelectElement>) =>
+    setCluster(event.target.value);
+  const selectNamespace = (event: React.ChangeEvent<HTMLSelectElement>) =>
+    setNS(event.target.value);
+  const changeContext = () => {
+    setNamespace(namespace);
+    dispatch(push(app.apps.list(cluster, namespace)));
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <div
+        className={`dropdown kubeapps-align-center kubeapps-dropdown ${open ? "open" : ""}`}
+        ref={ref}
+      >
+        <button className="kubeapps-nav-link" onClick={toggleOpen} aria-expanded={open}>
```

Comment:
Glad to see `aria-expanded` here :). I'm missing here the other `aria` property to link this button with the content: `aria-haspopup`. This property must use the same value as the `role` of the element with the content that it's placed next to the button:

```suggestion
        <button className="kubeapps-nav-link" onClick={toggleOpen} aria-expanded={open} aria-haspopup="menu">
```